AdminWebsite/AdminWebsite/ClientApp/src/app/services/video-hearings.service.ts

```diff
@@ -441,4 +443,20 @@ export class VideoHearingsService {
     getJudge(): ParticipantModel {
         return this.modelHearing.participants.find(x => x.is_judge);
     }
+
+    isConferenceClosed(): boolean {
+        this.checkForExistingHearing();
```

Comment:
Should this be asynchronous? i.e., use an obseravble?

Reply:
This is getting details from session storage so it's all local. I would like to see if there's a way this could be done once instead of every time either of these are called.

There's the following function: 

```
getCurrentRequest(): HearingModel {
        this.checkForExistingHearing();
        return this.modelHearing;
    }

```

Would it make sense to change this to the following and use that?

```
getCurrentRequest(): HearingModel {
        if(!this.modelHearing) {
                this.checkForExistingHearing();
        }
        return this.modelHearing;
    }

```

For reference checkForExistingHearing looks like this

```
    private checkForExistingHearing() {
        const localRequest = sessionStorage.getItem(this.newRequestKey);
        if (localRequest === null) {
            this.modelHearing = new HearingModel();
        } else {
            this.modelHearing = JSON.parse(localRequest);
        }
    }
```

Reply:
Yeah, I think that's a good suggestion - getting it from mem each time if it's there instead of from local storage. I'd also consider doing that when this service is initialised perhaps, and then just always use `this.modelHearing`?